test/integration/targets/flatpak/aliases

```diff
@@ -0,0 +1,7 @@
+unsupported
```

Comment:
@mattclay could you please check whether these tests are getting into CI correctly?

Reply:
From what I understand, the tests will not be run during normal CI runs triggered by pushes.

The reason for that I have chosen this is me being cautious: The tests depend on an external repository (Flathub) and on Ubuntu additionally on an external PPA (Flatpak). I am not sure how to evaluate the reliability of these resources from an automated CI perspective. I would not want to be responsible for the CI runs failing regularly because of Flathub answering slowly or the like. Not that I have experienced availability problems with Flathub or Launchpad.

Of course, the tests would deliver more value when being run automatically, so if you guys think the reliability bit is fine, I would be happy to have that changed.

Reply:
One way to do it is to use a local stub of the repos so that it'd hit them in CI.
Here's how we implemented a fake `deb` repo for `apt` module: #37639 + #38090.
Same for rpm: https://github.com/ansible/ansible/tree/devel/test/integration/targets/setup_rpm_repo

I believe you could try implementing something similar for your module. Makes sense?

Reply:
Sorry for the delay, I was at a conference this week.

Thank you for the suggestion and the links! I agree, this would be the best way to do this. I dug into into how to set up flatpak repos, it's doable, but I will need to add some functionality to the modules to turn off some features to support the dummy repo, like gpg-checking and dependency runtime installation. This could take some time (also because I will be travelling a whole month during autumn).

Unfortunately, right now we already feel the effect of not having any integration tests at all: Bugfix #44835 fixed Python 3 compatibility while breaking Python 2 compatibility. Something I now need to fix now asap.
If the current tests were available but unsupported, at least I could tell contributors how to easily run these to verify their contributions against regressions right now.

Do you think it would be reasonable to merge the integration tests as unsupported right now, giving me some leeway to properly implement integration tests with a dummy repository?

Reply:
I'm fine with whatever you decide. From the Core perspective, we don't care about this _too much_, because it's community-supported, but I still wanted to inform you about the options, because I know that it might be useful for you.

Reply:
@oolongbrothers I've merged this, so that it won't be hanging for undefined period of time and you could proceed with improvements whenever you have time :)

Reply:
That's great. Thanks, @webknjaz !

I'm indeed very thankful for your pointers towards using dummy repositories for testing. Once I get it implemented, it this will make the tests a lot faster and more reliable.

Could you also please merge #42315 which contains the integration tests for the sibling module `flatpak_remote`?